### ISSUE TITLE
Tensor FLoat 32 for all Ampere GPUs 

### ISSUE THREAD

**Issue by arnaud-nt2i** [AlexeyAB/darknet#7019]
As Stated here: https://github.com/AlexeyAB/darknet/issues/6802 and here https://developer.nvidia.com/blog/getting-immediate-speedups-with-a100-tf32/

Tensor FLoat 32 could leverage massive (x2 ~ x6) improvement in training speed without noticeable impact on accuracy.

TF32 is already implemented in TensorFlow and PyTorch, is there some plans to implement it in the Darknet framework?

**Comment by AlexeyAB** [comment]
Darknet already supports TF32 by default: https://github.com/AlexeyAB/darknet/blob/be906dfa0e1d24f5ba61963d16dd0dd00b32f317/src/convolutional_layer.c#L238

But for `mini_batch_size<=4` the A100 is only 2x times faster than V100.

So the most networks for visual tasks are only 2x times faster, while theoretical performance is 10x times faster:

* V100 - **14 TFlops (FP32)** - 112 TFlops (FP16) https://images.nvidia.com/content/technologies/volta/pdf/tesla-volta-v100-datasheet-letter-fnl-web.pdf
* A100 - **156 TFlopss (TF32)** - 312 TFlops (FP16) https://www.nvidia.com/content/dam/en-zz/Solutions/Data-Center/a100/pdf/nvidia-a100-datasheet.pdf

----

Seems to be only BERT can be greatly accelerated up to 6x times, while all other tasks can be accelerated only ~2x times (TF32 on NVIDIA 8xA100 vs. FP32 on 8xV100 GPU): https://developer.nvidia.com/blog/accelerating-tensorflow-on-a100-gpus/

![image](https://user-images.githubusercontent.com/4096485/96376324-ea5a2c00-1186-11eb-80ec-8f61e8d55823.png)

**Comment by arnaud-nt2i** [comment]
ok, Thank you For the quick answer.
Guess I need to wait for the end of training quietly ^^
